index.bs

```diff
@@ -359,14 +359,37 @@ Picture-in-Picture.
 
 # API # {#api}
 
-## Extensions to <code>HTMLVideoElement</code> ## {#htmlvideoelement-extensions}
+## Extensions to <code>HTMLElement</code> ## {#htmlelement-extensions}
 
 <xmp class="idl">
-partial interface HTMLVideoElement {
-  [NewObject] Promise<PictureInPictureWindow> requestPictureInPicture();
+partial interface HTMLElement {
+  [NewObject] Promise<PictureInPictureWindow> requestPictureInPicture(
+      PictureInPictureOptions options);
 
   attribute EventHandler onenterpictureinpicture;
   attribute EventHandler onleavepictureinpicture;
+};
+</xmp>
+
+The {{requestPictureInPicture(options)}} method, when invoked, MUST
+return <a>a new promise</a> |promise| and run the following steps <a>in
+parallel</a>:
+
+1. If <a dict-member for="PictureInPictureOptions">aspectRatio</a> is less than
+    or equal to zero then reject |promise| with <a exception>TypeError</a>.
+2. Let |element| be the element on which the method was invoked.
+3. Run the <a>request element Picture-in-Picture algorithm</a> with |element|
+    and <a dict-member for="PictureInPictureOptions">interactive</a>.
+4. If the previous step threw an exception, reject |promise| with that
+    exception and abort these steps.
+5. Return |promise| with the <a>Picture-in-Picture window</a> associated with
+    {{pictureInPictureElement}}.
+
+## Extensions to <code>HTMLVideoElement</code> ## {#htmlvideoelement-extensions}
+
+<xmp class="idl">
+partial interface HTMLVideoElement {
+  [NewObject] Promise<PictureInPictureWindow> requestPictureInPicture();
```

Comment:
Should we remove this and set `optional` to `PictureInPictureOptions options` in HTMLElement?
It looks like a video element will have both `requestPictureInPicture()` and `requestPictureInPicture(options)`. If I understood correctly, only the first one should work.

Reply:
Sounds good. Although in that case I think we should handle optional options on the element too. In which case we can populate the width/height from the current width/height of the element.